src/object.rs

```diff
@@ -1035,30 +1034,6 @@ macro_rules! glib_object_wrapper {
             }
         }
 
-        impl<T: $crate::object::ObjectType> ::std::cmp::PartialEq<T> for $name {
```

Comment:
This was actually needed :) It allows comparing e.g. a `gtk::Widget` to a `gtk::Window`. Now it's only possible to compare if they're of the same type.

Same thing below with `PartialOrd`.

This breaks applications now.